amqp/transport.py

```diff
@@ -506,20 +506,31 @@ def _wrap_socket_sni(self, sock, keyfile=None, certfile=None,
             'server_hostname': server_hostname,
         }
 
+        if ssl_version is None:
+            ssl_version = (
+                ssl.PROTOCOL_TLS_SERVER
+                if server_side
+                else ssl.PROTOCOL_TLS_CLIENT
+            )
+
         context = ssl.SSLContext(ssl_version)
+
         if certfile is not None:
             context.load_cert_chain(certfile, keyfile)
         if ca_certs is not None:
```

Comment:
If ca_certs is None but verify_mode is != CERT_NONE it is worth calling SSLContext.set_default_verify_paths() to attempt to load the system-wide CA files.

Reply:
I will address this in a different patch.